average.js

```diff
@@ -1,6 +1,21 @@
-
-function average(numbers) {
-  return numbers.reduce((p, c)=> p + c, 0) / numbers.length;
-}
-
-module.exports = {average};
+function average(numbers) {
+  let size = numbers.length;
+
+  if (size == 0) {
+    return NaN;
+  }
+
+  return (
+    numbers.reduce((p, c) => {
+      if (isNaN(c)) {

```

Comment:
You are doing the summing and elimination of NaN in a single pass. This is good in many situations.
However, consider the trade-off as well: Making the summing and counting NaNs in separate loops enables them to run in parallel, resulting in faster turnaround-time for large data-sets

Reply:
Created separate function for processing NaNs